definition_validator/validator.cc

```diff
@@ -156,8 +156,15 @@ pair<std::string, std::string> formatSplat(const core::ArgInfo &arg, SplatKind k
 
 optional<core::AutocorrectSuggestion> constructAllowIncompatibleAutocorrect(const core::Context ctx,
                                                                             const ast::ExpressionPtr &tree,
-                                                                            const ast::MethodDef &methodDef) {
-    if (!ctx.state.suggestUnsafe) {
+                                                                            const ast::MethodDef &methodDef,
+                                                                            bool &didReport) {
+    // XXX (cwong): From a UX perspective, this isn't great: with this design, we will suggest
+    // the autocorrection on the *first* error for a given override, whereas we'd ideally want to
+    // attach the autocorrect to the *last* error (where it's most likely to end up on the user's
+    // screen). I'm not super fussed about it in this case; `suggestUnsafe` is meant to be used
+    // with `-a` anyway, but if we decide to build out further support for "at most once"-type
+    // suggestions, it's something to keep in mind.
```

Comment:
fwiw, I think the UX is fine. In general with errors in Sorbet, the first errors are the most relevant, because later errors might only exist due to weird ways Sorbet attempted to recover from a later error. Also like you mention, the `--suggest-unsafe` nature of this means that you basically know what you're doing.